### ISSUE TITLE
Way to get Registry ACL

### ISSUE THREAD

**Issue by slaharip** [golang/go#26828]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
Go 1.10.3

### What operating system and processor architecture are you using (`go env`)?
GOHOSTARCH=amd64
GOHOSTOS=windows

### What did you do?
Using registry package to access the registry keys, values and their ACLs

If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
A link on play.golang.org is best.

### What did you expect to see?
NA

### What did you see instead?
NA

**Comment by ianlancetaylor** [comment]
I'm not sure but I think this is a feature request for golang.org/x/sys/windows/registry: provide some way to find the ACL of a key.  I don't know if that is meaningful or not.  If I'm wrong, please clarify.

**Comment by slaharip** [comment]
Yes, I am checking if there is/or isn't an equivalent of RegGetKeySecurity function in golang that retrieves the security descriptor protecting the specified open registry key.

**Comment by oryanmoshe** [comment]
If this is a wanted feature I can give it a look, seems like adding the winapi deceleration to `sys/windows/registry/syscall.go`, and adding GetACL and SetACL methods to `sys/windows/registry/key.go`

**Comment by alexbrainman** [comment]
> adding GetACL and SetACL methods to sys/windows/registry/key.go

I don't see how adding just 2 new methods will do here. What API addition did you have in mind?

Alex
